code/modules/surgery/organs/liver.dm

```diff
@@ -29,11 +29,13 @@
 				//handle liver toxin filtration
 				var/toxamount
 				var/static/list/listOfToxinsInThisBitch = typesof(/datum/reagent/toxin)
-				for(var/datum/reagent/toxin/toxin in listOfToxinsInThisBitch)
+				for(var/I in listOfToxinsInThisBitch)
+					var/datum/reagent/toxin = I
 					toxamount += C.reagents.get_reagent_amount(initial(toxin.id))
 
 				if(toxamount <= toxTolerance && toxamount > 0)
-					for(var/datum/reagent/toxin/toxin in listOfToxinsInThisBitch)
+					for(var/I in listOfToxinsInThisBitch)
+						var/datum/reagent/toxin = I //is this redundant because it's defined above?

```

Comment:
it's not redundant, the vars only exist within the scope they're defined in, so toxin won't exist at this scope level until you redefine it.

I would recommend renaming this var to `var/datum/reagent/the_toxin` or similar, otherwise it's clashing directly with the toxin typepath, that will still work but is confusing to people on first sight.

You could move the define up out of the for loops to avoid the duplication of code
```
var/datum/reagent/the_toxin = null
for (var/I in listofTo..)
      the_toxin = I
      ...
for (var/I in listofTo..)
      the_toxin = I
      ...
```

Reply:
Part of the issue is that listofToxins is a typelist rather than references.

Reply:
I'm so confused q-q
Why are we even defining var/datum/reagent/toxin/toxin? It doesn't get referenced anywhere.

Reply:
It gets referenced on line 39?

Reply:
What you're doing is basically changing from this:
`- Search for the elements of this specific type in this list, then iterate for each of them`
To this, which is generally faster:
`- Take every element of this list (which i know is only made of toxins), then iterate over each of them`
At this point the compiler does not treat the elements as toxins yet, so we define a variable with the correct type and we put the element in there. Now the compiler knows it's a toxin and can access its variables and functions without throwing an error.
When doing this, `var/datum/reagent/toxin` is the type, the following `toxin` is its name, which can be anything.
What oranges meant is that as you wrote it you were making a variable named `toxin` of type `datum/reagent`, instead of `datum/reagent/toxin`.

Reply:
Oh yeah it does.

And thanks for the explanation; it's a pretty confusing format for me.